### ISSUE TITLE
randomTTN for site-indices with QNs is missing

### ISSUE THREAD

**Issue by b-kloss** [mtfishman/ITensorNetworks.jl#124]
With the upcoming support for `QN` indices and fermions in the `TTN` constructor #122, it would be nice allow for making a random `TTN` in a given quantum number sector (determined by an input state, passed e.g. as a function).

**Comment by emstoudenmire** [comment]
I've been wanting to improve the implementation of this for the QN case ever since support for QN conserving QR (and LQ) decomposition was added. It would be good to chat about the algorithm for this & could be ideal to add it to this package then "back port" the implementation to ITensors.jl afterward.

**Comment by mtfishman** [comment]
Yeah, definitely something we need. Though it should be called `random_ttn` to follow the convention that only types (and constructors with the exact name of the type) are capitalized.

Also, a temporary workaround for constructing random TTN with bond dimension larger than one would be to apply a low-depth circuit of random 2-site unitary gates to a product state.
